### DISCUSSION TITLE
Does it make sense to add optional debug prints to from_json? Are there any better approaches?

### DISCUSSION THREAD

**Discussion by timyrka67** [nlohmann/json#2552] [Q&A] [answered]
There's difficulty to find a specific line of JSON file which causes a certain error. In the project, we have many structures with `adl_serializer` and nested `from_json`.

Example:
I have a  50k lines JSON file and somewhere there's an error:
`[json.exception.type_error.302] type must be number but is string`

So the quickest solution for me was adding try catches in sources:

```cpp
struct from_json_fn
{
    template<typename BasicJsonType, typename T>
    auto operator()(const BasicJsonType& j, T& val) const
    noexcept(noexcept(from_json(j, val)))
    -> decltype(from_json(j, val), void())
    {
      try
      {
        return from_json(j, val);
      }
      catch (const std::exception &ex)
      {
        std::cerr << "[NLOHMANN from_json error ]]: " << j << std::endl;
        throw;
      }
    }
};
```

Are there any better approaches for this case? The idea is to find a specific example with an error from the file

I would love to see something like [this](https://github.com/nlohmann/json/issues/1525#issuecomment-490462306):
```cpp
  try
  {
    // parsing input with a syntax error
    nlohmann::json::parse(myJsonFile);
  }
  catch (nlohmann::json::parse_error& e)
  {
    // output exception information
    std::cout << "message: " << e.what() << '\n'
              << "exception id: " << e.id << '\n'
              << "byte position of error: " << e.byte << std::endl;
  }
```
  But there isn't `byte` in `type_error`

**Comment by nlohmann** [comment, marked as answer]
FYI: I am trying to fix this issue in https://github.com/nlohmann/json/pull/2562. Any help would be greatly appreciated!

**Reply by nlohmann** [reply]
This is merged now. Do you need further assistance @timyrka67 ?
